geoips/interfaces/module_based/output_checkers.py

```diff
@@ -0,0 +1,557 @@
+                    if compare_strings is not None:
+                        for compare_string in compare_strings:
+                            badcomp = badcomp.replace(compare_string, "")
+                    self.print_gunzip_to_file(fobj, badcomp)
+                    fobj.write("cp -v {0} {1}/../\n".format(badcomp, diffdir))
+                    self.print_gzip_to_file(fobj, badcomp)
+            with open(fname_badcptest, "w") as fobj:
+                fobj.write("mkdir {0}/BADCOMPARES\n".format(diffdir))
+                for badcomp in badcomps:
+                    badcomp = badcomp.replace("IMAGE ", "")
+                    badcomp = badcomp.replace("TEXT ", "")
+                    badcomp = badcomp.replace("GEOIPS NETCDF ", "")
+                    badcomp = badcomp.replace("GEOTIFF ", "")
+                    # For display purposes - tifs are easier to view
+                    out_fname = basename(badcomp).replace(".jif", ".tif")
+                    self.print_gunzip_to_file(fobj, badcomp)
+                    fobj.write(
+                        "cp {0} {1}/BADCOMPARES/{2}\n".format(
+                            badcomp, diffdir, out_fname
+                        )
+                    )
+                    self.print_gzip_to_file(fobj, badcomp)
+
+        retval = 0
+        if len(badcomps) != 0:
+            curr_retval = len(badcomps)
+            if len(badcomps) > 4:
+                curr_retval = 4
+            retval += curr_retval << 0
+            LOG.info("BADCOMPS %s", len(badcomps))
+        if len(missingcomps) != 0:
+            curr_retval = len(missingcomps)
+            if len(missingcomps) > 4:
+                curr_retval = 4
+            retval += curr_retval << 0
+            LOG.info("MISSINGCOMPS %s", len(missingcomps))
+        if len(missingproducts) != 0:
+            curr_retval = len(missingproducts)
+            if len(missingproducts) > 4:
+                curr_retval = 4
+            retval += curr_retval << 0
+            LOG.info("MISSINGPRODUCTS %s", len(missingproducts))
+            retval += len(missingproducts) << 4
+
+        LOG.info("retval: %s", bin(retval))
+        if retval != 0:
+            LOG.info(
+                "Nonzero return value indicates error, 6 bit binary code: " "WXY where:"
+            )
+            LOG.info(
+                (
+                    "    Y (0-1): BADCOMP: number of bad comparisons found between"
+                    "comparepath and current run output path"
+                )
+            )
+            LOG.info(
+                (
+                    "    X (2-3): MISSINGCOMP: Number of products missing in "
+                    "compare path but existing in current output path"
+                )
+            )
+            LOG.info(
+                (
+                    "    W (4-5): MISSINGPROD: Number of products existing in "
+                    "comparepath, but missing in current output path"
+                )
+            )
+
```

Comment:
I think here at the very end you would have to re-zip all of the files found in "compare_path" as appropriate (so you may need to do some tracking of those in a dictionary or something)